### ISSUE TITLE
False errors being highlighted after 0.4.170 plugin update

### ISSUE THREAD

**Issue by Azgrom** [intellij-rust/intellij-rust#8845]
## Environment

* **IntelliJ Rust plugin version:** `0.4.170.5627-221`
* **Rust toolchain version:**
```sh
rustc 1.60.0 (7737e0b5c 2022-04-04)
rustup 1.24.3 (ce5817a94 2021-05-31)
cargo 1.60.0 (d1fd9fe2c 2022-03-01)
```
* **IDE name and version:**
```sh
 CLion 2022.1.1
Build #CL-221.5591.52, built on May 10, 2022
Runtime version: 11.0.14.1+1-b2043.45 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Windows 11 10.0
GC: G1 Young Generation, G1 Old Generation
Memory: 2000M
Cores: 12
Registry:
    run.processes.with.pty=TRUE
    ide.balloon.shadow.size=0

Non-Bundled Plugins:
    org.rust.lang (0.4.170.4627-221)
    com.github.lppedd.idea-conventional-commit (0.20.1)
    Key Promoter X (2022.1.2)
```
* **Operating system:**
```sh
Windows 11 Home Version 21H2
Build: 2000.675 
```

## Problem description
New version of Rust plugin points various errors on the same conditions `cargo check` points everything as **OK**.
![image](https://user-images.githubusercontent.com/14893177/168700220-6fe04b78-7dcb-41b1-a5c1-4d15cbfb54fd.png)

## Steps to reproduce

To reproduce, I believe that will be just [open the code](https://github.com/Azgrom/CarrierRoutes) with the specified CLion and Rust plugin.

**Comment by fujiapple852** [comment]
Can confirm, `0.4.170.4627-221` (note my version shows `4627` and not `5627` as mentioned by @Azgrom) on `2022.1.1` on macOS seems broken.  Gui stuck in a loop showing `Analysing...` and showing spurious errors.

**Comment by Undin** [comment]
@Azgrom @fujiapple852 It would be great to see your logs. See [these](https://intellij-support.jetbrains.com/hc/en-us/articles/207241085-Locating-IDE-log-files) instruction how to collect them

@Azgrom it looks like something happened with info about stdlib. As a result, the plugin doesn't know that primitive types implement `Copy` and produces `Cannot move` error. Try to [Invalidate caches](https://www.jetbrains.com/help/idea/invalidate-caches.html)

@fujiapple852 your error looks different. So it would be nice to get some code to reproduce your problem

**Comment by Azgrom** [comment]
Hey @Undin, invalidating the caches was enough for the errors stop being highlighted! 🤔

That was an on point hypothesis, thank you!

I believe this solves the question, and that the logs will not be necessary anymore, will they?

**Comment by fujiapple852** [comment]
@Undin after some fiddling (reinstalling the plugin and invalidating caches a few times) both the stdlib type checking errors and the weird `Analysing...` UX loop are resolved.

What I am left with are handful of errors that appear related to macro expansion, perhaps I should raise as a separate issue?  Anyway they mostly boil down to this case:

```rust
#[derive(Debug, derive_more::From)]
pub struct Foo(pub u16);

fn main() {
    let foo = Foo::from(0_u16);
    println!("{:?}", foo);
}
```

Which the IDE shows as:

<img width="500" alt="Screenshot 2022-05-18 at 9 47 47 AM" src="https://user-images.githubusercontent.com/36075719/168940465-e59c7c60-1eab-456f-b710-9afc00ae52d6.png">

Replacing the macro with its expansion and the IDE error is resolved:

```rust
impl ::core::convert::From<(u16)> for Foo {
    #[inline]
    fn from(original: (u16)) -> Foo {
        Foo(original)
    }
}
```

I observed similar with the `thiserror` derive `From` impls as well.

These cases worked in the previous version of the IDE ( `2021.3`) and the latest version of the plugin supported by that IDE version (unknown exact version).

**Comment by Azgrom** [comment]
## Update on the issue:

Since my last comment on this thread I had to invalidate cache 3 times because of a `moved value` error being erroneously highlighted...
